### PR TITLE
PYIC-7043 Add X-Forwarded-Host header

### PR DESCRIPTION
**Pull request by Joe-Edwards-GDS**
Adds an `X-Forwarded-Host` header so Dynatrace can detect the application

### CODE REVIEW

File: deploy/template.yaml [hunk 1]
```diff
@@ -1026,6 +1026,7 @@ Resources:
       RequestParameters:
         method.request.path.proxy: false
         method.request.header.x-forwarded-for: false
+        method.request.header.host: false
```
Comment (Joe-Edwards-GDS) [inline deploy/template.yaml:1029]: This defines `host` as a possible request header

File: deploy/template.yaml [hunk 2]
```diff
@@ -1036,6 +1037,7 @@ Resources:
         RequestParameters:
           integration.request.path.proxy: "method.request.path.proxy"
           integration.request.header.x-forwarded-for: "method.request.header.x-forwarded-for"
+          integration.request.header.x-forwarded-host: "method.request.header.host"
```
Comment (Joe-Edwards-GDS) [inline deploy/template.yaml:1040]: This adds an `x-forwarded-host` header on the response, with the value of the `host` header

File: deploy/template.yaml [hunk 3]
```diff
@@ -1003,7 +1003,7 @@ Resources:
         Types:
           - REGIONAL
 
-  RestApiGwDeployment202405141531:
+  RestApiGwDeployment202408070900:
```
Comment (Joe-Edwards-GDS) [inline deploy/template.yaml:1006]: Forces an API gateway deployment

**Review by vam-gds — APPROVED** [review]
Joe, Ive approved but make sure the headers required for txma are getting sent through still (the ip addresses / that they arent effected - this doesnt like it would do anything to them and should be fine)

**Pull request merged**